samples/oauth_react/src/Authentication.jsx

```diff
@@ -0,0 +1,143 @@
+// This module exports the Authentication element to manage the various
+// openid-client objects required to log in and out. It also exports a variety
+// of elements to make it easier to consume the authentication context without
+// having to care about AuthenticationContext.
+
+import React, { Component } from 'react';
+import { Log, UserManager } from 'oidc-client';
+
+const AuthenticationContext = React.createContext({user: null});
+
+class Authentication extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {authenticated: false, user: null};
```

Comment:
Looks like more legacy `authenticated` state stayed around?

Reply:
This one is intentional. We need to set the state to trigger the re-render after the `UserManager` callback executes. The `authenticated` field seems a redundant with the `user` field. However, the `user` field is set to whatever the `UserManager` passes to us. I didn't see anything in the documentation saying that it would never send us `null` so I wanted to make this eminently debuggable.